libs/ledger/src/miner/transaction_layout_queue.cpp

```diff
@@ -33,18 +33,15 @@ namespace ledger {
  */
 bool TransactionLayoutQueue::Add(TransactionLayout const &item)
 {
-  bool success{false};
-
   auto const &digest = item.digest();
 
-  // ensure that this isn't already a duplicate transaction layout
-  if (digests_.find(digest) == digests_.end())
+  // try to add this digest and check if it's new
+  bool success = digests_.insert(digest).second;
+
+  if (success)
   {
-    // update the digest set and list
+    // this is a non-duplicate transaction
```

Comment:
can you summarise the changes in this file? I don't fully understand what is going on